pkg/atomicwriter/atomic_writer_test.go

```diff
@@ -0,0 +1,243 @@
+package atomicwriter
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/stretchr/testify/assert"
+)
+
+type assertFunc func(path string, t *testing.T, err error)
+type errorAssertFunc func(buf *bytes.Buffer, wc io.WriteCloser, t *testing.T, err error)
+
+func TestWriteFile(t *testing.T) {
+	testCases := []struct {
+		name        string
+		path        string
+		permissions os.FileMode
+		content     string
+		assert      assertFunc
+	}{
+		{
+			name:        "happy path",
+			path:        "test_file.txt",
+			permissions: 0644,
+			content:     "test content",
+			assert: func(path string, t *testing.T, err error) {
+				assert.NoError(t, err)
+				// Check that the file exists
+				assert.FileExists(t, path)
+				// Check the contents of the file
+				contents, err := ioutil.ReadFile(path)
+				assert.NoError(t, err)
+				assert.Equal(t, "test content", string(contents))
+				// Check the file permissions
+				mode, err := os.Stat(path)
+				assert.NoError(t, err)
+				assert.Equal(t, os.FileMode(0644), mode.Mode())
```

Comment:
Coverage looks really good!

For completeness... would it be possible to check that the temp file is successfully deleted in the happy path? (Or maybe I'm missing where that's tested already?) I'm not sure if it's easier to add here (maybe return the `tempFile` in your `writeFile()` function, so it can be checked by the caller?), or some other happy path case.